internal/profile/profile.go

```diff
@@ -0,0 +1,265 @@
+	return profile.configFiles[file].FilePath
+}
+
+// UpdateFileBodies updates the string contents of the config files
+func (profile *ConfigProfile) updateFileBodies(newBody map[ConfigFile]*SimpleFile) {
+
+	for key := range profile.configFiles {
+		// skip metadata
+		if key == PackageProfileMetaFile {
+			continue
+		}
+		toReplace, ok := newBody[key]
+		if ok {
+			updatedProfile := profile.configFiles[key]
+			updatedProfile.FileBody = toReplace.FileBody
+			profile.configFiles[key] = updatedProfile
+		}
+	}
+
+}
+
+// ProfileAlreadyExists checks to see if a profile with this name already exists
+func (profile ConfigProfile) profileAlreadyExists() (bool, error) {
+
+	packageMetadata := profile.configFiles[PackageProfileMetaFile]
+	// We do this in stages to make sure we return the right error.
+	_, err := os.Stat(profile.ProfilePath)
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, errors.Wrapf(err, "error checking root directory: %s", packageMetadata.FilePath)
+	}
+
+	// If the folder exists, check to make sure it's a profile folder
+	_, err = os.Stat(packageMetadata.FilePath)
+	if os.IsNotExist(err) {
+		return false, ErrNotAProfile
+	}
+	if err != nil {
+		return false, errors.Wrapf(err, "error checking metadata: %s", packageMetadata.FilePath)
+	}
+
+	//if it is, see if it has the same profile name
+	profileInfo, err := profile.metadata()
+	if err != nil {
+		return false, errors.Wrap(err, "error reading metadata")
+	}
+
+	//TODO: this will break default_old, as we don't update the json
+	if profileInfo.Name != profile.profileName {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func (profile ConfigProfile) localFilesChanged() (bool, error) {
+
+	for cfgName, cfgFile := range profile.configFiles {
+		// skip checking the metadata file
+		// TODO: in the future, we might want to check version to see if the default profile needs to be updated
+		if cfgName == PackageProfileMetaFile {
+			continue
+		}
+		changes, err := cfgFile.ConfigfilesDiffer()
+		if err != nil {
+			return false, errors.Wrap(err, "error checking config file")
+		}
+		if changes {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// readProfileResources reads the associated files into the config, as opposed to writing them out.
+func (profile *ConfigProfile) readProfileResources() error {
+
+	for _, cfgFile := range profile.configFiles {
+		err := cfgFile.ReadConfig()
+		if err != nil {
+			return errors.Wrap(err, "error reading in profile")
+		}
+	}
+
+	return nil
+}
+
+// writeProfileResources writes the config files
+func (profile ConfigProfile) writeProfileResources() error {
+
+	for _, cfgFiles := range profile.configFiles {
+		err := cfgFiles.WriteConfig()
+		if err != nil {
+			return errors.Wrap(err, "error writing config file")
+		}
+	}
+	return nil
+}
+
+// metadata returns the metadata struct for the profile
+func (profile ConfigProfile) metadata() (Metadata, error) {
```

Comment:
WDYT about returning nil instead of plain Metadata in case of an error?

Reply:
We'd need to return `*Metadata`, so I'm not sure it makes that much of a difference, considering we don't really need to return a pointer for any particular reason, unless I'm missing something.